include/kernel.h

```diff
@@ -1320,13 +1320,24 @@ __syscall void *k_thread_custom_data_get(void);
 /* kernel clocks */
 
 #ifdef CONFIG_SYS_CLOCK_EXISTS
-#if	(CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC % sys_clock_ticks_per_sec) != 0
+
+/*
+ * If timer frequency is known at compile time, a simple (32-bit)
+ * tick <-> ms conversion could be used for some combinations of
+ * hardware timer frequency and tick rate. Otherwise a precise
+ * (64-bit) calculations are used.
+ */
+
+#if !defined(CONFIG_TIMER_READS_ITS_FREQUENCY_AT_RUNTIME)
+#if	(sys_clock_hw_cycles_per_sec % sys_clock_ticks_per_sec) != 0
```

Comment:
Just a suggestion:
sys_clock_hw_cycles_per_sec is a macro in this case; still, better to use CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC, directly, as people are more comfortable with seeing "real" macro names in #ifdef statements.

Reply:
I used `sys_clock_hw_cycles_per_sec` on purpose, as this is real source of the information. It is either define (and could be used at compile time) or a variable. In some configuration it just equals to CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC.